x/register/keeper/querier.go

```diff
@@ -78,3 +101,357 @@ func GetNetworkSet(ctx sdk.Context, k Keeper) ([]byte, error) {
+	resourceResultsLen := len(resourceNodeResults)
+	start, end := client.Paginate(indexingResultsLen+resourceResultsLen, params.Page, params.Limit, QueryDefaultLimit)
+	var bz []byte
+	if start < 0 || end < 0 {
+		return bz, nil
+	}
+	if end <= indexingResultsLen {
+		bzIndexing, err := codec.MarshalJSONIndent(keeper.cdc, indexingNodeResults[start:end])
+		if err != nil {
+			return nil, sdkerrors.Wrap(sdkerrors.ErrJSONMarshal, err.Error())
+		}
+		bz = append(bz, bzIndexing...)
+	} else if start > indexingResultsLen {
+		bzResource, err := codec.MarshalJSONIndent(keeper.cdc, resourceNodeResults[start:end])
+		if err != nil {
+			return nil, sdkerrors.Wrap(sdkerrors.ErrJSONMarshal, err.Error())
+		}
+		bz = append(bz, bzResource...)
+	} else {
+		bzIndexing, err := codec.MarshalJSONIndent(keeper.cdc, indexingNodeResults[start:indexingResultsLen])
+		if err != nil {
+			return nil, sdkerrors.Wrap(sdkerrors.ErrJSONMarshal, err.Error())
+		}
+		bz = append(bz, bzIndexing...)
+
+		bzResource, err := codec.MarshalJSONIndent(keeper.cdc, resourceNodeResults[:end-indexingResultsLen])
+		if err != nil {
+			return nil, sdkerrors.Wrap(sdkerrors.ErrJSONMarshal, err.Error())
+		}
+		bz = append(bz, bzResource...)
+	}
+	return bz, nil
+}
+
+func (k Keeper) GetIndexingNodesFiltered(ctx sdk.Context, params QueryNodesParams) []types.IndexingNode {
+	nodes := k.GetAllIndexingNodes(ctx)
+	filteredNodes := make([]types.IndexingNode, 0, len(nodes))
+
+	for _, n := range nodes {
+		// match NetworkID (if supplied)
+		if len(params.NetworkID) > 0 {
+			if strings.Compare(n.NetworkID, params.NetworkID) != 0 {
+				continue
+			}
+		}
+
+		// match Moniker (if supplied)
+		if len(params.Moniker) > 0 {
+			if strings.Compare(n.Description.Moniker, params.Moniker) != 0 {
+				continue
+			}
+		}
+
+		// match OwnerAddr (if supplied)
+		if params.OwnerAddr.Empty() || n.OwnerAddress.Equals(params.OwnerAddr) {
+			filteredNodes = append(filteredNodes, n)
+		}
+	}
+	filteredNodes = k.indPagination(filteredNodes, params)
+	return filteredNodes
+}
+
+func (k Keeper) GetResourceNodesFiltered(ctx sdk.Context, params QueryNodesParams) []types.ResourceNode {
+	nodes := k.GetAllResourceNodes(ctx)
+	filteredNodes := make([]types.ResourceNode, 0, len(nodes))
+
+	for _, n := range nodes {
+		// match NetworkID (if supplied)
+		if len(params.NetworkID) > 0 {
+			if strings.Compare(n.NetworkID, params.NetworkID) != 0 {
+				continue
+			}
+		}
+
+		// match Moniker (if supplied)
+		if len(params.Moniker) > 0 {
+			if strings.Compare(n.Description.Moniker, params.Moniker) != 0 {
+				continue
+			}
+		}
+
+		// match OwnerAddr (if supplied)
+		if params.OwnerAddr.Empty() || n.OwnerAddress.Equals(params.OwnerAddr) {
+			filteredNodes = append(filteredNodes, n)
+		}
+	}
+
+	filteredNodes = k.resPagination(filteredNodes, params)
+	return filteredNodes
+}
+
+func (k Keeper) resPagination(filteredNodes []types.ResourceNode, params QueryNodesParams) []types.ResourceNode {
```

Comment:
give full name of resource node

Reply:
rename function resPagination to resourceNodesPagination